### ISSUE TITLE
Provide an action to unmark branches as finished

### ISSUE THREAD

**Issue by lf-** [ubc-carnap-team/Rudolf#113]
It seems to be impossible to undo the actions "Close Branch with Contradiction" and "Mark Branch Finished" without the undo button. Case where this is annoying: user just got most of the way through a proof and realize they made a mistake.

I can't find a right click menu on anything to do this, so if it is possible, I just can't find it.

![image](https://user-images.githubusercontent.com/6652840/126855903-d337658e-836c-4631-a489-08fd1f2c4c0a.png)

There is no right click handler on the X here, and there seems to be no action on line 10 to undo it either.
![image](https://user-images.githubusercontent.com/6652840/126855937-c08106d7-19b6-47d8-acf9-0a60477bfaa6.png)
![image](https://user-images.githubusercontent.com/6652840/126855955-90bca92f-b54c-432c-bfb4-8f6eb1784443.png)

**Comment by McTano** [comment]
You're right. The only way to reopen a branch now is with undo, which is inconvenient for the reasons you stated.
Shouldn't be too hard to implement now. I'll take care of it.

**Comment by McTano** [comment]
Now that I'm working on this, I remember why we ended up leaving this feature unimplemented.
The problem is that if you close a branch and then reopen it later, that branch might be missing some nodes that would have been created if it had been open.

e.g. say we split to resolve P&Q, then close the left branch before resolving R&S. If we reopen the left branch, we'll need to add the node with the formulas `R,S` again to complete the tree.
```
    P\/Q
    R&S
     /   \
    P   Q
    x    R
          S
```

That's why it's currently necessary to use undo and do the work again.

I think it should still be possible for the student to reopen a branch and redo the missing work manually on that branch. I'll test it out and make sure the checker accepts it.
